### PR TITLE
Msgpack update

### PR DESCRIPTION
**Pull request by PierrePharel**

### CODE REVIEW

File: msgpack/include/adaptor.hpp [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+			{
+				msgpack::object const& operator()(msgpack::object const& o, Value& v) const 
+				{
+					if(o.type == type::NIL)
+					{
+						v = Value(-1);
+					}
+					else if(o.type == type::BOOLEAN)
+					{
+						v = Value(o.via.boolean);
```
Comment (SuperFola) [inline msgpack/include/adaptor.hpp:24]: won't work with the new api, we don't have a constructor accepting booleans

File: msgpack/include/adaptor.hpp [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+						{
+							Value ark_object_v;
+
+							if(o.via.array.ptr[i].type == type::NIL)
+							{
+								ark_object_v = Value(-1);
+							}
+							else if(o.via.array.ptr[i].type == type::BOOLEAN)
+							{
+								ark_object_v = Value(o.via.array.ptr[i].via.boolean);
```
Comment (SuperFola) [inline msgpack/include/adaptor.hpp:48]: same here for the boolean

File: msgpack/src/object.cpp [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+	Value print_msgpack_o(std::vector<Value> &args)
+	{
+		if(args.size() != 1)
+			throw std::runtime_error("ArgError : This function must have 1 argument");
+		msgpack::object* o {static_cast<msgpack::object*>(args[0].usertype().data())};
+
+		std::cout << "msgpack::object : " << *o << std::endl;
+
+		return Value(Ark::Nil);
```
Comment (SuperFola) [inline msgpack/src/object.cpp:79]: return Ark::Nil should work too and would avoid a copy

File: msgpack/src/object.cpp [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+		if(args.size() != 1)
+			throw std::runtime_error("ArgError : This function must have 1 argument");
+		if(args[0].valueType() != ValueType::User || args[0].usertype().type_id() != std::type_index(typeid(msgpack::object_handle)))
+			throw Ark::TypeError("The packed buffer must be a msgpack::object_handle");
+		static msgpack::object o;
+
+		msgpack::object_handle* oh {static_cast<msgpack::object_handle*>((args[0]).usertype().data())};
+		o = oh->get();
+
+		return Value(UserType(&o));
```
Comment (SuperFola) [inline msgpack/src/object.cpp:68]: we can have only 1 msgpack object at a time?

Comment (PierrePharel) [reply msgpack/src/object.cpp:68]: Yes ! One obejct for one pack process

File: msgpack/src/object.cpp (hunk 3 above)
Comment (PierrePharel) [reply msgpack/src/object.cpp:79]: Ok

File: msgpack/include/adaptor.hpp (hunk 2 above)
Comment (PierrePharel) [reply msgpack/include/adaptor.hpp:48]: Ok

File: msgpack/include/adaptor.hpp (hunk 1 above)
Comment (PierrePharel) [reply msgpack/include/adaptor.hpp:24]: Ok

**Pull request merged**